internal/pgadmin/users_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+ Copyright 2021 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package pgadmin
+
+import (
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"gotest.tools/v3/assert"
+
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+)
+
+func TestWriteUsersInPGAdmin(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("Arguments", func(t *testing.T) {
+		expected := errors.New("pass-through")
+		exec := func(
+			_ context.Context, stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			assert.Assert(t, stdin != nil, "should send stdin")
+			assert.Assert(t, stdout != nil, "should capture stdout")
+			assert.Assert(t, stderr != nil, "should capture stderr")
+
+			assert.Check(t, !strings.ContainsRune(strings.Join(command, ""), '\t'),
+				"Python should not be indented with tabs")
+
+			assert.DeepEqual(t, command, []string{"python3", "-c", `
```

Comment:
:thinking: Is there a linter we can run Python code through, e.g. like we do with our `bash` code using `shellcheck`?

Reply:
I know of `pylint` and `flake8`, and I see CentOS packages for both. I'll try them out.

Reply:
Sounds good!  Definitely not needed for this specific PR, but figure a linter could help any non-Python coders keep these parts of the code clean (I'm also happy to add an item to the backlog for this, so just let me know).

Reply:
[Pylint expects to run modules](https://pylint.pycqa.org/en/latest/user_guide/run.html). It wants a comment describing the module. The assignments are top-level, so it wants them spelled like constants.

[Flake8](https://github.com/pycqa/flake8) has fewer complaints, so I used that and addressed issues it raised. Neither tool comes pre-installed in the [Linux provided by GitHub Actions](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md). We'll need to do more if we want that test to run there.

```console
$ pylint script.py
************* Module script
script.py:4:0: C0301: Line too long (101/100) (line-too-long)
script.py:1:0: C0114: Missing module docstring (missing-module-docstring)
script.py:1:0: C0410: Multiple imports on one line (importlib.util, os, sys) (multiple-imports)
script.py:3:0: C0103: Constant name "spec" doesn't conform to UPPER_CASE naming style (invalid-name)
script.py:4:0: C0103: Constant name "root" doesn't conform to UPPER_CASE naming style (invalid-name)
…
```

Reply:
Excellent, that sounds great!